sdk/typescript/src/transactions/TransactionBlockDataResolver.ts

```diff
@@ -98,21 +93,24 @@ export class TransactionBlockDataResolver {
 		args: Parameters<TransactionBlockDataResolver[T]>,
 		cb: () => ReturnType<TransactionBlockDataResolver[T]>,
 	): Promise<ReturnType<TransactionBlockDataResolver[T]>> {
-		for (const plugin of this.plugins) {
-			if (method in plugin) {
-				const result = await (
-					plugin[method as never] as (
-						...args: unknown[]
-					) => ReturnType<TransactionBlockDataResolver[T]>
-				)(this, ...args);
+		const methods = this.plugins
+			.filter((plugin) => method in plugin)
+			.map((plugin) => plugin[method]!);
 
-				if (result != null) {
-					return result;
-				}
+		function run(index: number, runArgs: unknown[]) {
```

Comment:
Why not reduce instead of doing it recursively like this?

Reply:
because this is doing something a little more complicated.

If you have a data resolver like:
```ts
{
  getObjects: (ids, next) => {
     const existing = cache.get(ids)
     const missing = ids.filter(id => !existing.find(id))
     const newObjects = next(ids)
     ....
    }
 }
 ```
 
You are calling `next` with a different set of ids than was originally passed in, and could potentially call next multiple times.  The final results are also fed back up into the previous resolver before it returns, so you can't really do a reduce.